authenticator/nonce_validator.go

```diff
@@ -0,0 +1,43 @@
+package authenticator
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"strconv"
+	"time"
+)
+
+const maxTimeGap = 30000 // 30 secs
+
+//ErrNonceNotInRange error when nonce not in aceptable range
+var ErrNonceNotInRange = errors.New("nonce submit is not in aceptable range")
+
+// NonceValidator checking validate by time range
+type NonceValidator struct {
+	// TimeGap is max time different between client submit timestamp
+	// and server time that considered valid. The time precision is millisecond.
+	TimeGap uint64
+}
+
+// NewNonceValidator return NonceValidator with default value (30 second)
+func NewNonceValidator() *NonceValidator {
```

Comment:
I think this function should have param to modify TimeGap

Reply:
> I think this function should have param to modify TimeGap

did that in ff05ebf1faab1d43286f9cabed7e9ac5547723ea.